src/osd/PG.cc

```diff
@@ -5576,7 +5578,20 @@ void PG::handle_query_state(Formatter *f)
   recovery_state.handle_event(q, 0);
 }
 
-
+void PG::update_store_with_options()
+{
+  bufferlist bl;
+  pool.info.opts.encode(bl);
+  __u32 crc = bl.crc32c(0);
+  if (crc != last_pool_opts_crc) {
+    ObjectStore::Transaction t;
+    t.set_collection_opts(coll, bl);
+    auto r = osd->store->queue_transaction(osr.get(), std::move(t), nullptr);
+    if (r==0){
+      last_pool_opts_crc = crc;
+    }
+  }
+}
```

Comment:
we can also avoid the bufferlist encode/decode and just pass pool_opts_t by reference...